File: Metode-Avansate-de-Programare/lab2/model/MessageTask.java
package sem1_2.model;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task {

    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    public static DateTimeFormatter getDateFormatter() {
        return dateFormatter;
    }

    private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    public MessageTask(String taskId, String description, String message, String from, String to, LocalDateTime date) {
        super(taskId, description);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    @Override
    public void execute() {
        System.out.println(date.format(dateFormatter) + ": " + message);
    }

    @Override
    public String toString() {
        return "id = " + getId() +
                " | description = " + getDescription() +
                " | message = " + message +
                " | from = " + from +
                " | to = " + to +
                " | date = " + date.format(dateFormatter);
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/MessageTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

public class MessageTask : Task
{
    private string message;
    private string from;
    private string to;
    private DateTime date;

    public MessageTask(string taskID, string description, string message, string from, string to, DateTime date) : base(
        taskID, description)
    {
        this.message = message;
        this.from = from;
        this.date = date;
    }

    public override void execute()
    {
        //Console.WriteLine(taskID + " " + description + " " + from + " " + message + " " + date);
    }

    public string Message
    {
        get { return message; }
        set { message = value; }
    }

    public string To
    {
        get { return to; }
        set { to = value; }
    }

    public string From
    {
        get { return from; }
        set { from = value; }
    }

    public DateTime Date
    {
        get { return date; }
        set { date = value; }
    }
}